cmd/resource-topology-exporter/main.go

```diff
@@ -41,176 +51,103 @@ func main() {
-			[--topology-manager-policy=<pol>]
-			[--reference-container=<spec>]
-			[--config=<path>]
-
-  {{.ProgramName}} -h | --help
-  {{.ProgramName}} --version
-
-  Options:
-  -h --help                       Show this screen.
-  --debug                         Enable debug output. [Default: false]
-  --version                       Output version and exit.
-  --no-publish                    Do not publish discovered features to the
-                                  cluster-local Kubernetes API server.
-  --hostname                      Override the node hostname.
-  --oneshot                       Update once and exit.
-  --sleep-interval=<seconds>      Time to sleep between podresources API polls.
-                                  [Default: 60s]
-  --export-namespace=<namespace>  Namespace on which update CRDs. Use "" for all namespaces.
-  --watch-namespace=<namespace>   Namespace to watch pods for. Use "" for all namespaces.
-  --sysfs=<path>                  Top-level component path of sysfs. [Default: /sys]
-  --kubelet-config-file=<path>    Kubelet config file path. [Default: ]
-  --topology-manager-policy=<pol> Explicitely set the topology manager policy instead of reading
-                                  from the kubelet. [Default: ]
-  --kubelet-state-dir=<path>...   Kubelet state directory (RO access needed), for smart polling.
-  --podresources-socket=<path>    Pod Resource Socket path to use.
-                                  [Default: unix:///podresources/kubelet.sock]
-  --reference-container=<spec>    Reference container, used to learn about the shared cpu pool
-                                  See: https://github.com/kubernetes/kubernetes/issues/102190
-                                  format of spec is namespace/podname/containername.
-                                  Alternatively, you can use the env vars
-                                  REFERENCE_NAMESPACE, REFERENCE_POD_NAME, REFERENCE_CONTAINER_NAME.
-  --config=<path>                 Configuration file path. Use this to set the exclude list.
-                                  [Default: /etc/resource-topology-exporter/config.yaml]`
-
-func getUsage() (string, error) {
-	var helpBuffer bytes.Buffer
-	helpData := struct {
-		ProgramName string
-	}{
-		ProgramName: ProgramName,
-	}
+type config struct {
+	ExcludeList map[string][]string
+}
 
-	tmpl, err := template.New("help").Parse(helpTemplate)
+func readConfig(configPath string) (config, error) {
+	conf := config{}
+	data, err := os.ReadFile(configPath)
 	if err != nil {
-		return "", err
+		// config is optional
+		if errors.Is(err, os.ErrNotExist) {
+			log.Printf("Info: couldn't find configuration in %q", configPath)
+			return conf, nil
+		}
+		return conf, err
 	}
-	err = tmpl.Execute(&helpBuffer, helpData)
-	if err != nil {
-		return "", err
+	err = yaml.Unmarshal(data, &conf)
+	return conf, err
+}
+
+// The args is passed only for testing purposes.
+func parseArgs(args ...string) (*progArgs, error) {
```

Comment:
why do we need to return a pointer to `progArgs`?

Reply:
Avoid redundant copy.

Reply:
Which happens once however. Do we have stronger reasons?
http://macias.info/entry/201802102230_go_values_vs_references.md
https://medium.com/@philpearl/bad-go-pointer-returns-340f2da8289

Reply:
Cool articles, wasn't aware of that. Will fix

Reply:
sure, my point here is that using pointer values instead of copying values is not obviously better every time, and it has its own fair share of drawbacks. So it should be a conscious decision when do what. I for myself I believe there is unwarranted fear about returning and passing by "large" values. We should not just default to "use pointer, it's faster".